File: gateway/src/main/java/com/undancer/app/BarController.java
package com.undancer.app;


import com.google.common.collect.Maps;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;

@RestController
@RequestMapping(path = "/bar")
public class BarController {

    @GetMapping
    public Object home() {
        Map<String, Object> resultObject = Maps.newHashMap();
        resultObject.put("foo", "bar");
        return resultObject;
    }

}

File: gateway/src/main/kotlin/com/undancer/cloud/gateway/BarController.kt
package com.undancer.cloud.gateway


import com.google.common.collect.Maps
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(path = arrayOf("/bar"))
class BarController {

    @GetMapping
    fun home(): Any {
        val resultObject = Maps.newHashMap<String, Any>()
        resultObject["foo"] = "bar"
        return resultObject
    }

}
